src/client/d3/d3.js

```diff
@@ -218,20 +224,38 @@ function render(updatedData) {
     if ($('.delete-button').hasClass('deleting')) {
       const id = d3.select(this).attr('id');
       $('.delete-button').removeClass('deleting');
-      function update(url) {
+      function update() {
         d3.json(url, (err, data) => {
           render(formatData(data));
-          sim.restart();
         });
       }
       $.ajax({
         method: 'DELETE',
         url: 'memories',
         data: { id },
-        success: () => update(url),
+        success: () => update(),
       });
     }
     hideDeleteButton();
-    sim.restart();
+  }
+
+  $('#memory-input__submit').click((e) => {
+    e.preventDefault();
+    $.ajax({
+      method: 'POST',
+      url: 'memory-input-text',
+      data: $('#add-text-form').serialize(),
+      success: () => {
+        setTimeout(() => {
+          update();
+        }, 4500);
```

Comment:
I don't think it's a hack necessarily... when we add a node, we have a 4.5 second animation so I wanted to wait for that to finish before re rendering the map